### DISCUSSION TITLE
Add extra person with required fields toe booking

### DISCUSSION THREAD

**Discussion by Wessiez** [liedekef/events-made-easy#477] [General]
Goodday,

We have an event where parents add bookings for their childeren. I'm making a template for participating in this event.
But some parents have 1 child, others have 3 children. How can I make a template where the parent can add more childeren (persons) with required field.

Example 1 (parent with 1 child): 
Required_Name Parent:
Required_Name Child 1:
Required_medal-number Child 1:

Example 2 (parent with 3 children): 
Required_Name Parent:
Required_Name Child 1:
Required_medal-number Child 1:
Required_Name Child 2:
Required_medal-number Child 2:
Required_Name Child 3:
Required_medal-number Child 3:

I cannot figure this out. 
I hope that there is an solution with a "add child" button.
Does anybody knows how to make it work?

**Comment by liedekef** [comment]
See the "Dynamic data" section here: https://www.e-dynamics.be/wordpress/registration-form-formatting/

**Reply by Wessiez** [reply]
Is there a tutorial or an example of using the #_DYNAMICDATA placeholder?

Is er ergens een Tutorial of een voorbeeld hoe men de #_DYNAMICDATA placeholder gebruikt?

**Reply by liedekef** [reply]
Yes, exactly on that link. Just try it on a test event ...

**Reply by Wessiez** [reply]
I still don't get it.

![image](https://github.com/liedekef/events-made-easy/assets/5537264/1e1af9dd-724b-4935-b0bc-120317ce56f0)
What does the "field"contains? what does the "condition value" contains? 
And the template is a template with the name of the Child and the choosen medalnumber?

Template:
![image](https://github.com/liedekef/events-made-easy/assets/5537264/d7411bad-4e15-4da7-b7e5-7486f838b1d6)

**Reply by liedekef** [reply]
The field-param is the field that will be compared against the mentioned condition. For example: "#_FIELD{children}" (where that field is a custom created field asking for the number of children) and the condition (in your case) should be "equal or greater than" and as value probably 1. Then the template used (with "repeat" set to "yes") should contain extra custom field placeholders asking for the name/... of 1 child. Due to setting "repeat" to "yes", that template will then be used for info on each child.
If you don't know how to work with custom fields, then first play with those before going for dynamic data

**Reply by Wessiez** [reply]
Hello Franky,

First of all I like to thank you for your patience with me!

I managed to implement the dynamic data. I ask the question if they want to add more children. Standard in the dropdown menu is "No" and the numbers show the extra template. So far so good.

For this event can be choosen for a medal with a number or no medal at all.
When they choose a medal they have to pay €6,00. But when the participate without a medal the price is €2,00.
#_SEATS{1} = €6,00
#_SEATS{2} = €2,00
How can this be intergraded in the dynamic data so it adds up when the payment can be done?

**Reply by liedekef** [reply]
Create a custom field of type dropdown, set "extra price" to "yes", set as "field values" the price values for each (negative prices are accepted, so counting as discounts) and as "field tags" the names you want ('Medal', 'No medal'). The "field values" should be unique though, so watch out for that

**Reply by Wessiez** [reply]
I tried the above, but it doesn't work:

I set the RSVP info with the 2 prices en give them names:
![image](https://github.com/liedekef/events-made-easy/assets/5537264/6b197e3f-7d11-4dd8-b4b5-53458d3b9044)

This is the booking_template:

```
<div class="wpex-alert wpex-alert-info">- Meedoen is op eigen risico.
- Indien uw kind niet alleen kan lopen, moet er begeleiding mee.
- De "aanmelder" is degene die betaalt en op de hoogte wordt gehouden.
- De "wandelaars" zijn degene die daadwerkelijk wandelen. Dus eventueel ook de aanmelder.</div>
<div class="form-section">
<h3>Gegevens Aanmelder</h3>
<fieldset><legend> </legend>
<div style="text-align: left;"><label style="background-color: #b8deff; display: block; padding: 5px 10px; border-radius: 5px;" for="parent_first_name">Voornaam aanmelder: #REQ_FIRSTNAME</label><span style="display: block; margin-top: 10px;">
</span></div>
<div><label style="background-color: #b8deff; display: block; padding: 5px 10px; border-radius: 5px;" for="parent_last_name">Achternaam aanmelder: #REQ_LASTNAME</label><span style="display: block; margin-top: 10px;">
</span></div>
<div><label style="background-color: #b8deff; display: block; padding: 5px 10px; border-radius: 5px;" for="parent_email">E-mail aanmelder: #REQ_EMAIL</label><span style="display: block; margin-top: 10px;">
</span></div>
<div><label style="background-color: #b8deff; display: block; padding: 5px 10px; border-radius: 5px;" for="parent_phone">Telefoonnummer aanmelder: #REQ_PHONE</label><span style="display: block; margin-top: 10px;">
</span></div>
<div>

<label style="background-color: #b8deff; display: block; padding: 5px 10px; border-radius: 5px;" for="parent_phone">Lid worden appgroep: #REQ_FIELD{22}</label>
<div class="wpex-alert wpex-alert-info">T.b.v. het snel kunnen sturen van updates, voortgang &amp; foto's is het mogelijk om je aan te melden voor de (SJD) WhatsApp groep van de Wandel4Daagse.</div>
<div></div>
</div></fieldset>
</div>
[vcex_divider color="#dddddd" width="100%" height="1px" margin_top="20" margin_bottom="20"]
<h3>Gegevens Wandelaar(s)</h3>
<fieldset><legend> </legend>
<div style="text-align: left;"><label style="background-color: #b8deff; display: block; padding: 5px 10px; border-radius: 5px;" for="parent_first_name">Hoeveel wandelaars worden er opgeven? #_FIELD{A4D_aantal_wandelaars}</label></div></fieldset>
&nbsp;

#_SUBMIT{Betaal en reserveer}
<div>
<em>U zult automatisch worden doorgestuurd naar de iDeal pagina.</em>

```

The template for adding attendies:
```
<div class="form-section">
<h3>Wandelaar</h3>
<div><label style="background-color: #b8deff; display: block; padding: 5px 10px; border-radius: 5px;" for="child_first_name">Voornaam:  #REQ_FIELD{9}</label><span style="display: block; margin-top: 10px;">
</span></div>
<div><label style="background-color: #b8deff; display: block; padding: 5px 10px; border-radius: 5px;" for="child_last_name">Achternaam:  #REQ_FIELD{10}</label><span style="display: block; margin-top: 10px;">
</span></div>
<div><label style="background-color: #b8deff; display: block; padding: 5px 10px; border-radius: 5px;" for="child_school">School:  #REQ_FIELD{13}</label><span style="display: block; margin-top: 10px;">
</span></div>
<div>
<div class="wpex-alert wpex-alert-info">Geef het gewenste medaillenummer op. Het nummer op de medaille geeft aan voor de hoeveelste keer er dit jaar meegelopen wordt.</div>
</div>
<div>

<label style="background-color: #b8deff; display: block; padding: 5px 10px; border-radius: 5px;" for="child_school">Een medaille ontvangen: #REQ_FIELD{23} </label>

<label style="background-color: #b8deff; display: block; padding: 5px 10px; border-radius: 5px;" for="child_school">Nummer op de medaille: #REQ_FIELD{20}</label>

</div>
</div>
```
The custom field for choosing medal or no medal:
![image](https://github.com/liedekef/events-made-easy/assets/5537264/c9ae3b89-8863-4b42-a6be-87e03cdc427c)

And the message when I test this all:
![image](https://github.com/liedekef/events-made-easy/assets/5537264/b898ad5d-bf90-4ef1-9dd2-47881dd580a5)


What goes wrong here?

**Reply by liedekef** [reply]
If you have multiple price categories, then you need to let people chose from multiple seat categories too. If you want that, your booking template needs to contain #_SEATS{1}, #_SEATS{2} etc.
Next to that, your "veldwaarden" can't contain placeholders, only actual values (prices in this case), but those are not needed if you work with multiple price categories then. It all depends on how you want the total price to be calculated.
Lastly: while I'm here to answer how EME works, I'm not here to answer single use-case questions. Those fall outside this support channel (and are not free). In your case you first need to familiarize yourself with dynamic data and custom fields in general.
While EME can do a lot, you need to play with it first if you have special needs for your event.

**Reply by Wessiez** [reply]
> If you have multiple price categories, then you need to let people chose from multiple seat categories too. If you want that, your booking template needs to contain #_SEATS{1}, #_SEATS{2} etc.
> Next to that, your "veldwaarden" can't contain placeholders, only actual values (prices in this case), but those are not needed if you work with multiple price categories then. It all depends on how you want the total price to be calculated.

I will look at this. Maybe I need to change to 1 price in the RSVP and give a discount for No medal in the custom field were the choice of a medal is made...

> Lastly: while I'm here to answer how EME works, I'm not here to answer single use-case questions. Those fall outside this support channel (and are not free). In your case you first need to familiarize yourself with dynamic data and custom fields in general.
> While EME can do a lot, you need to play with it first if you have special needs for your event.
> 
I'm sorry for that Franky. I thought this was the forum. My apologies...
